[ipv6] AntreaProxy IPv6 support

ksamoray opened the pull request:
Implement IPv6 dual stack support for Antrea Proxy

ksamoray commented:
/test-e2e

wenyingd commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -534,6 +538,10 @@ func (c *client) connectionTrackFlows(category cookie.Category) []binding.Flow {
+				Action().CT(false, connectionTrackTable.GetNext(), CtZoneV6).CTDone().
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
 			connectionTrackCommitTable.BuildFlow(priorityLow).MatchProtocol(binding.ProtocolIP).
```
This flow has the corresponding flow for IPv6, is it as expected?

ksamoray replied:
You refer to connectionTrackCommitTable flow? Yeah it seems like I'm missing an IPv6 flow.

wenyingd commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -1493,18 +1507,43 @@ func (c *client) serviceLearnFlow(groupID binding.GroupIDType, svcIP net.IP, svc
 	} else if protocol == binding.ProtocolSCTP {
 		learnFlowBuilder.MatchSCTPDstPort(svcPort)
 		learnFlowBuilderLearnAction = learnFlowBuilderLearnAction.MatchLearnedSCTPDstPort()
+	} else if protocol == binding.ProtocolTCPv6 {
```
Use switch-case instead of so many if-else if ?

ksamoray replied:
OK

wenyingd commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -1517,6 +1556,12 @@ func (c *client) serviceLBFlow(groupID binding.GroupIDType, svcIP net.IP, svcPor
 		lbFlowBuilder = lbFlowBuilder.MatchUDPDstPort(svcPort)
 	} else if protocol == binding.ProtocolSCTP {
 		lbFlowBuilder = lbFlowBuilder.MatchSCTPDstPort(svcPort)
+	} else if protocol == binding.ProtocolTCPv6 {
```
Use switch-case?

wenyingd commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -1531,33 +1576,57 @@ func (c *client) serviceLBFlow(groupID binding.GroupIDType, svcIP net.IP, svcPor
-		LoadToMark(serviceCTMark).
-		CTDone().
-		Done()
+	if ipProtocol == binding.ProtocolIP {
```
Since most parts of the flow are the same, what about change to this

```
flowBuilder := table.BuildFlow(priorityNormal).
		Cookie(c.cookieAllocator.Request(cookie.Service).Raw()).
		MatchRegRange(int(endpointPortReg), unionVal, binding.Range{0, 18}).
		MatchProtocol(ipProtocol)
if ipProtocol == binding.ProtocolIP {
    ipVal := binary.BigEndian.Uint32(endpointIP.To4())
    flowBuilder = flowBuilder.MatchReg(int(endpointIPReg), ipVal).
		MatchRegRange(int(endpointPortReg), unionVal, binding.Range{0, 18})
} else {
    ipVal := []byte(endpointIP)
    flowBuilder = flowBuilder.MatchXxReg(int(endpointIPv6XXReg), ipVal).
		MatchRegRange(int(endpointPortReg), unionVal, binding.Range{0, 18}).
}
return flowBuilder.Action().CT(true, table.GetNext(), CtZone).
		DNAT(
				&binding.IPRange{StartIP: endpointIP, EndIP: endpointIP},
				&binding.PortRange{StartPort: endpointPort, EndPort: endpointPort},
		).
		LoadToMark(serviceCTMark).
		CTDone().
		Done()
```

ksamoray replied:
Indeed much cleaner.

wenyingd commented on pkg/agent/proxy/meta_proxier.go:
```diff
@@ -0,0 +1,182 @@
+// SyncLoop runs periodic work.  This is expected to run as a
+// goroutine or as the main loop of the app.  It does not return.
+func (proxier *metaProxier) SyncLoop() {
+	go proxier.ipv6Proxier.SyncLoop() // Use go-routine here!
```
Why use goroutine for ipv6Proxier, but not for ipv4Proxier?

ksamoray replied:
SyncLoop() itself is called using goroutine. So ipv4proxier resume in parent context (and runs indefinitely), while ipv6proxier spawns from a new context.

wenyingd replied:
In my opinion, the process for IPv4 and IPv6 should be similar. If leaving ipv4Proxier in the parent context just to ensure the function to run infinitely, we could use a waitGroup. No strong preferrence.

ksamoray replied:
Using a waitGroup will retain one extra context which will pend forever as SyncLoop runs indefinitely, right?

wenyingd reviewed (commented):
Sorry for not finish all code, will continue tomorrow.

ksamoray commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -261,6 +262,8 @@ var (
 	// the selected Service Endpoint IP.
 	endpointIPRegRange = binding.Range{0, 31}
+	// the selected Service Endpoint IPv6.
+	endpointIPv6XXRegRange = binding.Range{0, 128}
```
Indeed. Thanks.

weiqiangt commented on pkg/agent/proxy/proxier.go:
```diff
@@ -117,13 +112,23 @@ func (p *proxier) removeStaleServices() {
-		}
+		var bindingProtocol binding.Protocol
 		for _, endpoint := range endpoints {
+			if utilnet.IsIPv6String(endpoint.IP()) {
```
I suggest extracting this big if block to a separate function to improve readability. The go compiler will inline the function I guess, then there should be no different.

weiqiangt commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -218,6 +218,7 @@ const (
 	// cnpDropConjunctionIDReg reuses reg3 which will also be used for storing endpoint IP to store the rule ID. Since
 	// the service selection will finish when a packet hitting NetworkPolicy related rules, there is no conflict.
 	cnpDropConjunctionIDReg regType = 3
+	endpointIPv6XXReg       regType = 3 // xxReg3 will occupy 12-15 regs
```
I think we'd better introduce a new type here to avoid confusion. @wenyingd What do you think?

ksamoray replied:
Could be. But then we risk someone reusing regs 12-15 here, overwriting the xxreg.

tnqn replied:
I'm not sure what "xxReg3 will occupy 12-15 regs" here? xxReg3 will occupy reg12-reg15?

ksamoray replied:
Indeed. Will change the comment.

ksamoray commented:
Hi @wenyingd @weiqiangt 
I've mostly based meta_proxier.go on K8S v1.17 [ipvs meta_proxier](https://github.com/kubernetes/kubernetes/blob/release-1.17/pkg/proxy/ipvs/meta_proxier.go).
One option is to add waitGroup to meta_proxier funcs, divert further from the original implementation.
Other option is to move meta_proxier.go to third_party/proxy, document the differences as we did with the other third_party modules (and while at this maybe upgrade to a later version of K8S proxy code?).
LMK what you think.

wenyingd commented:
> Hi @wenyingd @weiqiangt
> I've mostly based meta_proxier.go on K8S v1.17 [ipvs meta_proxier](https://github.com/kubernetes/kubernetes/blob/release-1.17/pkg/proxy/ipvs/meta_proxier.go).
> One option is to add waitGroup to meta_proxier funcs, divert further from the original implementation.
> Other option is to move meta_proxier.go to third_party/proxy, document the differences as we did with the other third_party modules (and while at this maybe upgrade to a later version of K8S proxy code?).
> LMK what you think.

Personally, I don't think it is necessary to put it under third_party, no strong preferrence. As it is code for Antrea Proxy, and I think you just follow the design for meta_proxier in ipvs, we could place it under pkg/... In my mind, most code under third_party/... are copied from the upstreaming directly (not sure about it), @weiqiangt might know more details.

ksamoray commented:
/test-all

ksamoray commented:
/test-ipv6-e2e

ksamoray commented:
/test-ipv6-conformance
/test-ipv6-only-e2e
/test-ipv6-only-conformance

wenyingd commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -725,8 +740,8 @@ func (c *client) traceflowL2ForwardOutputFlow(dataplaneTag uint8, category cooki
 // hairpin packets to avoid packets from being dropped by OVS.
-func (c *client) l2ForwardOutputServiceHairpinFlow() binding.Flow {
-	return c.pipeline[L2ForwardingOutTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+func (c *client) l2ForwardOutputServiceHairpinFlow(ipProtocol binding.Protocol) binding.Flow {
```
As this flow doesn't match any IP or ct_mark, I am thinking is it possible to remove the match condition of "ip" or "ipv6", then we could trate the packet as a L2 packet, and only one flow entry could satisfy the requirement.

ksamoray replied:
Wouldn't that affect other protocols e.g arp? IDK if that has any importance as I think that these packets are treated earlier in the chain.

wenyingd replied:
I don't think so. As ARP packets would be forwarded in table20, it can't go ino this table.

ksamoray replied:
Make sense. I'll apply this change. Thanks!

wenyingd commented on pkg/ovs/openflow/ofctrl_action.go:
```diff
@@ -379,18 +390,36 @@ func (a *ofLearnAction) MatchLearnedTCPDstPort() LearnAction {
+// MatchLearnedUDPv6DstPort specifies that the udp_dst field in the learned flow
+// must match the udp_dst of the packet currently being processed.
+func (a *ofLearnAction) MatchLearnedUDPv6DstPort() LearnAction {
+	return a.MatchTransportDst(ProtocolUDPv6)
```
Please take care of the implementation of function `MatchTransportDst` (https://github.com/vmware-tanzu/antrea/blob/master/pkg/ovs/openflow/ofctrl_action.go#L363), I remembered there should be some hardcode for IP protocol. If you plan to use the same function, you need to ensure that function is also available for IPv6 protocols.

ksamoray replied:
Done

wenyingd commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -1477,39 +1498,72 @@ func (c *client) serviceLearnFlow(groupID binding.GroupIDType, svcIP net.IP, svc
-		Action().LoadRegRange(int(serviceLearnReg), marksRegServiceSelected, serviceLearnRegRange).
-		Action().GotoTable(endpointDNATTable).
-		Done()
+	ipProtocol := getIPProtocol(svcIP)
```
I think the variable `ipProtocol` could be set value in the above switch-case according to the given protocol?

ksamoray replied:
Done

ksamoray commented on pkg/agent/openflow/client.go:
```diff
@@ -404,7 +404,7 @@ func (c *client) InstallEndpointFlows(protocol binding.Protocol, endpoints []pro
-		endpointIP := net.ParseIP(endpoint.IP()).To4()
+		endpointIP := net.ParseIP(endpoint.IP())
 		portVal := uint16(endpointPort)
 		cacheKey := fmt.Sprintf("Endpoints:%s:%d:%s", endpointIP, endpointPort, protocol)
```
Functionally that won't have any impact - in the end it's just a string which will be unique regardless of IP protocol support. Maybe will make this more readable if we ever dump the cache contents.
I'll wait for @weiqiangt feedback about this.

weiqiangt replied:
Sorry, I have just noticed it. @ksamoray, you're right, the name format has no special limit.
To make it more readable, I'm OK about using `_`.

ksamoray replied:
Thanks :)

ksamoray commented:
/test-all

ksamoray commented:
/test-ipv6-conformance
/test-ipv6-e2e
/test-ipv6-networkpolicy
/test-ipv6-only-conformance
/test-ipv6-only-e2e
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-all
/test-ipv6-conformance
/test-ipv6-e2e
/test-ipv6-networkpolicy
/test-ipv6-only-conformance
/test-ipv6-only-e2e
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-whole-conformance

ksamoray commented:
/test-conformance-coverage

ksamoray commented:
/test-windows-networkpolicy
/test-windows-conformance

ksamoray commented:
/test-e2e-coverage

ksamoray commented:
/test-whole-conformance-coverage
/test-networkpolicy-coverage
/test-ipv6-ds-conformance

ksamoray commented:
/test-conformance-coverage
/test-ipv6-only-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance

ksamoray commented:
/test-ipv6-ds-conformance
/test-ipv6-ds-e2e
/test-ipv6-only-e2e

ksamoray commented:
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-whole-conformance-coverage

ksamoray commented:
/test-all
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-only-conformance
/test-ipv6-only-e2e

ksamoray commented:
/test-e2e
/test-ipv6-e2e

ksamoray commented:
/test-ipv6-only-conformance
/test-ipv6-only-e2e
/test-networkpolicy-coverage
/test-ipv6-only-networkpolicy

tnqn commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -254,6 +255,8 @@ var (
 	// the selected Service Endpoint IP.
 	endpointIPRegRange = binding.Range{0, 31}
+	// the selected Service Endpoint IPv6.
+	endpointIPv6XXRegRange = binding.Range{0, 127}
```
We should keep the same convention of writing XXReg in the whole patch (some code use XxReg)

ksamoray replied:
OK

tnqn commented on pkg/agent/openflow/pipeline.go:
```diff
@@ -527,13 +531,24 @@ func (c *client) connectionTrackFlows(category cookie.Category) []binding.Flow {
 				Cookie(c.cookieAllocator.Request(category).Raw()).
 				Done(),
+			connectionTrackTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIPv6).
+				Action().CT(false, connectionTrackTable.GetNext(), CtZoneV6).CTDone().
```
just to confirm, the NAT action is note added by intention as IPv6 doesn't need NAT?

ksamoray replied:
Service cluster IP still NATs the pod IP, even with IPv6. Right?

tnqn replied:
that's my question, I'm not sure the answer. If we need to NAT cluster IP for IPv6 as well, why we don't have "NAT()" action here?

ksamoray replied:
Indeed. Thanks for noticing

tnqn commented on pkg/ovs/openflow/ofctrl_action.go:
```diff
@@ -127,8 +130,16 @@ func (a *ofCTAction) natAction(isSNAT bool, ipRange *IPRange, portRange *PortRan
+		} else {
+			action.SetRangeIPv4Min(ipRange.StartIP)
+		}
+		if utilnet.IsIPv6(ipRange.EndIP) {
```
I think ipRange won't mix protocol family, so L139 can be after L134?

ksamoray replied:
Indeed

tnqn commented on test/integration/agent/openflow_test.go:
```diff
@@ -266,7 +266,7 @@ func testInstallTunnelFlows(t *testing.T, config *testConfig) {
 
 func testInstallServiceFlows(t *testing.T, config *testConfig) {
-	err := c.InstallClusterServiceFlows()
+	err := c.InstallClusterServiceFlows(true, true)
```
out of curiosity, with IPv6 introduced, why there's no change to the expected output of openflow in the test?

ksamoray replied:
Service cluster IP still NATs the pod IP, even with IPv6. Right?

ksamoray replied:
Service cluster IP still NATs the pod IP, even with IPv6. Right?

ksamoray replied:
Where the test does validate the IPs... I didn't change the original tests functionality, but modified the existing ones to cover IPv6 when the IPs were validated.

lzhecheng commented:
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-networkpolicy

ksamoray commented on build/yamls/antrea.yml:
```diff
@@ -1035,7 +1035,7 @@ data:
     # Service traffic. Antrea proxy doesn't support an IPv6 only cluster or a Dual-Stack cluster
     # before PR #1102[https://github.com/vmware-tanzu/antrea/pull/1102] is merged.
-    #  AntreaProxy: false
+      AntreaProxy: true
```
Indeed. No CI for IPv6+AntreaProxy as of now, so activated it here to have a test cycle on CI

ksamoray commented on pkg/agent/proxy/meta_proxier.go:
```diff
@@ -0,0 +1,182 @@
+// Copyright 2020 Antrea Authors
```
Initially it was and I proposed having this as 3rd party above. Then with the changes to the Provider interface, I'm not sure that it still make sense.

tnqn replied:
If only a few lines are changed, I think it should be still kept under 3rd and kept its original copyright, we could comment what we changed like @weiqiangt did: https://github.com/vmware-tanzu/antrea/blob/master/third_party/proxy/endpoints.go#L16-L38

ksamoray replied:
OK

ksamoray commented on pkg/agent/proxy/proxier.go:
```diff
@@ -74,6 +68,7 @@ type proxier struct {
 	stopChan     <-chan struct{}
 	agentQuerier querier.AgentQuerier
 	ofClient     openflow.Client
+	enableIPV6   bool
```
OK

ksamoray commented:
/test-all

ksamoray commented:
/test-e2e-coverage
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-all
/test-e2e-coverage
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-windows-networkpolicy

ksamoray commented:
/test-e2e

ksamoray commented:
/test-ipv6-e2e

ksamoray commented:
/test-conformance-coverage
/test-e2e-coverage

ksamoray commented:
/test-e2e-coverage
/test-conformance-coverage

ksamoray commented:
/test-ipv6-conformance
/test-ipv6-e2e
/test-ipv6-only-conformance
/test-ipv6-only-e2e

ksamoray commented:
/test-conformance
/test-ipv6-conformance
/test-ipv6-only-conformance

ksamoray commented:
/test-all

ksamoray commented:
/test-ipv6-e2e
/test-ipv6-only-e2e
/test-ipv6-conformance
/test-ipv6-only-conformance

ksamoray commented:
/test-ipv6-e2e
/test-e2e-coverage
/test-conformance-coverage

ksamoray commented:
/test-e2e

ksamoray commented:
/test-all-features-conformance

ksamoray commented:
/test-all
/test-whole-conformance
/test-all-features-conformance

ksamoray commented:
/test-conformance

ksamoray commented:
/test-all-features-conformance

ksamoray commented:
/test-e2e

ksamoray commented:
/test-ipv6-e2e
/test-ipv6-only-e2e

ksamoray commented:
/test-e2e-coverage

ksamoray commented:
/test-all
/test-whole-conformance

ksamoray commented:
/test-ipv6-e2e
/test-ipv6-only-e2e

ksamoray commented:
/test-conformance

ksamoray commented:
/test-conformance

ksamoray commented:
/test-all

ksamoray commented:
/test-ipv6-e2e
/test-ipv6-only-e2e

ksamoray commented:
/test-e2e-coverage

lzhecheng commented:
/test-ipv6-e2e
/test-ipv6-only-e2e

lzhecheng commented:
/test-ipv6-e2e

lzhecheng commented:
e2e cases seem running. If there's any CI issue, I will solve them tomorrow.
/test-ipv6-conformance
/test-ipv6-only-conformance

lzhecheng commented:
/test-ipv6-only-conformance

lzhecheng commented:
/test-ipv6-e2e

lzhecheng commented:
/test-ipv6-e2e

salv-orlando commented:
/test-all-features-conformance

ksamoray commented:
/test-e2e-coverage

ksamoray commented:
/test-ipv6-only-conformance

ksamoray commented:
/test-ipv6-e2e

ksamoray commented:
/test-ipv6-only-e2e

ksamoray commented:
/test-ipv6-conformance
/test-e2e-coverage
/test-all-features-conformance

ksamoray commented:
/test-ipv6-conformance
/test-e2e-coverage
/test-all-features-conformance
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-e2e-coverage

ksamoray commented:
/test-conformance-coverage

ksamoray commented:
/test-whole-conformance-coverage

ksamoray commented:
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-ipv6-e2e

ksamoray commented:
/test-whole-conformance-coverage
/test-windows-networkpolicy
/test-networkpolicy-coverage
/test-ipv6-networkpolicy

ksamoray commented:
/test-e2e-coverage

ksamoray commented:
/test-conformance

ksamoray commented:
/test-conformance-coverage

ksamoray commented:
/test-whole-conformance-coverage

ksamoray commented:
/test-ipv6-only-e2e
/test-ipv6-e2e

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-ipv6-only-e2e

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

tnqn reviewed (approved):
LGTM, a comment about verbose log.

Hope @weiqiangt can review this as he knows AntreaProxy best.

weiqiangt commented on pkg/agent/openflow/client.go:
```diff
@@ -407,9 +408,12 @@ func (c *client) InstallEndpointFlows(protocol binding.Protocol, endpoints []pro
 		var flows []binding.Flow
 		endpointPort, _ := endpoint.Port()
-		endpointIP := net.ParseIP(endpoint.IP()).To4()
+		endpointIP := net.ParseIP(endpoint.IP())
```
Could we ensure the endpoints all have the same IP family? I think we can validate the IP family out of the loop.
Furthermore, if the endpoints all have the same IP family, can we pass the IP family as an argument explicitly instead of checking it?

weiqiangt replied:
Sorry, I should point to the next line. I just feel we can avoid that 'heavy' checking.

ksamoray replied:
Done

weiqiangt commented on third_party/proxy/types.go:
```diff
@@ -29,20 +29,36 @@ limitations under the License.
 
 import (
 	"fmt"
+	"github.com/vmware-tanzu/antrea/third_party/proxy/config"
```
wrong import grouping

ksamoray replied:
Indeed. Thanks.

ksamoray commented on pkg/ovs/openflow/ofctrl_action.go:
```diff
@@ -430,6 +471,19 @@ func (a *ofLearnAction) MatchReg(regID int, data uint32, rng Range) LearnAction
+// MatchXXReg makes the learned flow to match the data in the xxreg of specific range.
+func (a *ofLearnAction) MatchXXReg(regID int, data []byte, rng Range) LearnAction {
+	s := fmt.Sprintf("NXM_NX_XXREG%d", regID)
+	klog.Infof("LearnAction MatchXXReg s=%s", s)
```
This log is redundant - debugging leftover... Thanks for noticing.

lzhecheng commented:
Hello, will you rebase and set `master` branch as target? I do this to my PR as well.

ksamoray commented:
> Hello, will you rebase and set `master` branch as target? I do this to my PR as well.

Yes, while addressing the review comments. Is the ipv6 branch merged already?

lzhecheng commented:
> > Hello, will you rebase and set `master` branch as target? I do this to my PR as well.
> 
> Yes, while addressing the review comments. Is the ipv6 branch merged already?

Yes, it was merged several hours ago.

weiqiangt reviewed (dismissed):
LGTM

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

ksamoray commented:
/test-e2e
/test-conformance
/test-networkpolicy
/test-windows-conformance
/test-windows-networkpolicy
/test-ipv6-e2e
/test-ipv6-conformance
/test-ipv6-networkpolicy
/test-ipv6-only-e2e
/test-ipv6-only-conformance
/test-ipv6-only-networkpolicy

tnqn reviewed (approved):
LGTM

ksamoray commented:
go test -v github.com/vmware-tanzu/antrea/test/e2e -provider=kind

ksamoray commented:
/test-e2e-coverage

ksamoray commented:
/test-ipv6-e2e

ksamoray commented:
/test-ipv6-e2e
